core/src/impl/Kokkos_Profiling.cpp

```diff
@@ -118,15 +118,17 @@ bool eventSetsEqual(const EventSet& l, const EventSet& r) {
          l.request_output_values == r.request_output_values &&
          l.declare_optimization_goal == r.declare_optimization_goal;
 }
-template <typename Callback, typename BooleanConstant, typename... Args>
-inline void invoke_kokkos_callback(const Callback& callback, BooleanConstant,
-                                   Args&&... args) {
+enum class MayRequireGlobalFencing : bool { Yes, No };
```

Comment:
Minor: I would have done `{ No, Yes }`, mainly, so that `No` matches `false` and `Yes` matches `true`.

Reply:
Done in 2791e3f